ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/part/explorer/project/NewProjectExplorerPresenter.java

```diff
@@ -163,19 +170,25 @@ public void onResourceEvent(ResourceNodeDeletedEvent evt) {
         eventBus.addHandler(ResourceNodeRenamedEvent.getType(), new ResourceNodeRenamedEvent.ResourceNodeRenamedHandler() {
             @Override
             public void onResourceRenamedEvent(final ResourceNodeRenamedEvent event) {
-                HasDataObject dataObject = new HasDataObject() {
-                    @NotNull
-                    @Override
-                    public Object getData() {
-                        return event.getNewDataObject();
-                    }
-
-                    @Override
-                    public void setData(@NotNull Object data) {
-
-                    }
-                };
-                reloadChildren(event.getNode().getParent(), dataObject);
+                Object newDataObject = event.getNewDataObject();
+                String path = null;
+
+                if (newDataObject instanceof ProjectReference) {
```

Comment:
Need to simplify if possible or add some comments because hard to understand this block